### PR TITLE
Adds in pydantic settings to easily make interfaces to services. v2

### LINKED ISSUE #388: Consider pydantic settings managment
We could consider making a pydantic settings class to cover all of the settings [here](https://github.com/choderalab/asapdiscovery/blob/a4bda414a8920ca161e303399e2ae4ce3026db82/asapdiscovery-docking/asapdiscovery/docking/scripts/dock_small_scale_end_to_end.py#L390-L422) as I am sure other work flows may need to reuse these like uploading the results of FEC? Pydantic can handle gathering the settings from the environment automatically see [here](https://docs.pydantic.dev/2.0/usage/pydantic_settings/#settings-management). We can then add validation methods which checks they are all set.

### PR DESCRIPTION
**Pull request by hmacdope**
Fixes #388  

Superseeds #400 which requires pydantic 2.0 +

### CODE REVIEW

**Comment by hmacdope** [comment]
@jthorton could I grab a review here?

File: asapdiscovery-docking/asapdiscovery/docking/scripts/dock_small_scale_end_to_end.py [hunk 1]
```diff
@@ -385,47 +392,17 @@ def main():
...
-            raise ValueError("Environment variable AWS_SECRET_ACCESS_KEY not found")
-
-        artifact_bucket_name = os.getenv("ARTIFACT_BUCKET_NAME")
-        if artifact_bucket_name is None:
-            raise ValueError("Environment variable ARTIFACT_BUCKET_NAME not found")
-
-        cloudfront_domain = os.getenv("CLOUDFRONT_DOMAIN")
-        if cloudfront_domain is None:
-            raise ValueError("Environment variable CLOUDFRONT_DOMAIN not found")
+        postera_settings = PosteraSettings()
```
Comment (jthorton) [inline asapdiscovery-docking/asapdiscovery/docking/scripts/dock_small_scale_end_to_end.py:395]: Do we want to check that all of the settings can be found earlier in the workflow so we fail fast by this point we have already started creating folders.

File: asapdiscovery-data/asapdiscovery/data/services_config.py [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+    AWS_ACCESS_KEY_ID: str
+    AWS_SECRET_ACCESS_KEY: str
+    BUCKET_NAME: str
+    BUCKET_PREFIX: Optional[str] = None
+
+
+class CloudfrontSettings(BaseSettings):
+    CLOUDFRONT_DOMAIN: str
+    CLOUDFRONT_KEY_ID: str
+    CLOUDFRONT_PRIVATE_KEY_PEM: str
```
Comment (jthorton) [inline asapdiscovery-data/asapdiscovery/data/services_config.py:23]: Consider adding fields so we can have a description of what should go here this is the only one that is not clear by name alone to me.

**Review by jthorton — CHANGES_REQUESTED** [review]
Nice glad this is working. I would defiantly add a test for the validator the other feedback Ill leave to your choice :).

File: asapdiscovery-docking/asapdiscovery/docking/scripts/dock_small_scale_end_to_end.py (hunk 1 above)
Comment (hmacdope) [reply asapdiscovery-docking/asapdiscovery/docking/scripts/dock_small_scale_end_to_end.py:395]: Good point

**Comment by hmacdope** [comment]
@jthorton could I get a re-review here?

**Review by jthorton — APPROVED** [review]
LGTM! great job testing everything as well.

**Pull request merged**